dingo/gw/waveform_generator/waveform_generator.py

```diff
@@ -3,24 +3,35 @@
 from math import isclose
 
 import numpy as np
+import astropy.units as u
 from typing import Dict, List, Tuple, Union
 from numbers import Number
 import warnings
 import lal
 import lalsimulation as LS
 import pandas as pd
+
+try:
+    from lalsimulation.gwsignal.core import waveform as gws_wfm
+    from lalsimulation.gwsignal.models import (
+        gwsignal_get_waveform_generator as new_interface_get_waveform_generator,
+    )
+except:
+    print(
+        "LALsimulation does not contain gwsignal module, you will not be able to employ new interface."
+    )
```

Comment:
Error message deleted, but imports couldn't be included inside init method, otherwise they are not accessible outside, and storing the imported methods as attributes produces a pickle error in the code. I've tested and with an environment that does not contain latest lalsimulation, it will just not import that, without errors